src/server/common/helpers/redis-client.js

```diff
@@ -0,0 +1,65 @@
+import { Redis } from 'ioredis'
+
+import { createLogger } from './logging/logger.js'
+
+import config from '~/src/server/config.js'
+
+/**
+ * Setup Redis and provide a redis client
+ *
+ * Local development - 1 Redis instance
+ * Out in the wild - Elasticache / Redis Cluster with username and password
+ * @returns {import('ioredis').Cluster | import('ioredis').Redis}
+ */
+export function buildRedisClient() {
+  const logger = createLogger()
+
+  const port = 6379
+  const db = 0
+  let redisClient
+
+  if (config.env !== 'production') {
+    logger.info('Connecting to Redis using single instance')
+
+    redisClient = new Redis({
+      username: config.redisUsername,
+      password: config.redisPassword,
+      port,
+      host: config.redisHost,
+      db
+    })
+  } else {
+    logger.info('Connecting to Redis using cluster')
+
+    redisClient = new Redis.Cluster(
+      [
+        {
+          host: config.redisHost,
+          port
+        }
+      ],
+      {
+        slotsRefreshTimeout: 2000,
+        dnsLookup: (address, callback) => callback(null, address),
+        redisOptions: {
+          username: config.redisUsername,
+          password: config.redisPassword,
+          db,
+          tls: {}
+        }
+      }
+    )
+  }
+
+  // TODO add proper logger
+
+  redisClient.on('connect', () => {
+    logger.info('Connected to Redis server')
+  })
+
+  redisClient.on('error', (error) => {
+    logger.error(error, `Redis connection error ${error}.`)
+  })
```

Comment:
Do we want to keep "connection closed" logging from **forms-designer**?

```suggestion
  redisClient.on('close', () => {
    logger.info('Redis connection closed')
  })

  redisClient.on('error', (error) => {
    logger.error(error, `Redis connection error ${error}.`)
  })
```

Reply:
It will log by default so we don't need it fortunately.

If we override the `close` event listener it will disable the re-try logic.

Reply:
Brill, thanks for confirming

Reply:
Correction: I am mistaken on this, something else I was working on disabled the re-try logic, so I've restored the logger as a `warn` event.